src/widgets/panel/panel.tsx

```diff
@@ -81,13 +81,13 @@ export type PanelWidgetOptions = {
    * A function that is called on each render to determine if the
    * panel should be hidden based on the render options.
    */
-  hidden?(options: RenderOptions): boolean;
+  hidden?(options: RenderOptions & { widgetRenderState?: unknown }): boolean;
```

Comment:
Not sure what we can do about this. Any idea?

Reply:
this is an alternative where panel is generic and you need to repeat the connector type, that might be good enough:

https://www.typescriptlang.org/play?#code/C4TwDgpgBA6glgEwOYWAHgCoD4oF4oDeAUFFCsPMqgEoQB2CEATgMrACGwEAFAJQBcUDAG4iAX1FFQkKAGEA9nToQAxsHlNMOfHzw5K5LZIBmAVzpq4iqGHbKANpigQAHlwYBnOYuVqNaOxAsLG4SGzsIewAFdiZ2AFsPQWJSUgALREY6blikJMIw1KgAd0RyWgZmNk4IQVpgUyY6DHAINHrG5tatAG0AcnIDGnpGVg4uPoBdLFEisQEoQNmoMQAaIl4C0iZUTqhuFR9VdSZBDE3cHAIxcSIiQ7oPYCh4kAUlY41Bd98TtAIoId5PZBE8mHA6EgVtp9hccNwUmRUENgBVRtUuLpEdtdk1CID5MDBH0AEYQOwePorZZrcS8SS2BxoaQQeTGF5vI5+JghMLYqAZBBZHJMPKbfmkB4eYEQAB09nkSBFYuWpFpavWvG4rx+nyYvCIQA

Reply:
https://github.com/algolia/instantsearch.js/pull/4527/commits/de2cbef68588d7209893db7038bf5f616d740680
it works!

```
panel<BreadcrumbWidget>({ ... })(breadcrumb);
```

if I don't provide the generic, then `widgetRenderState` is `any`.